edx-solutions/edx-platform
# Automatic provisioning of account for SAML users

**JIRA Ticket:** https://openedx.atlassian.net/browse/YONK-80
**PR deployment instructions:** contains a migration - make sure to run `./manage.py lms migrate --settings=devstack` or equivalent
**Testing instructions:**
1. Follow instructions at [Shibboleth Configuration](https://openedx.atlassian.net/wiki/display/OpenOPS/New+Shibboleth+Configuration) docs until step 6.
2. The easiest way to set up a second part of SAML authentication process is to use http://testshib.org
2.1. Export edX SSO metadata to a file.
2.2. Import them at ["Register" step](https://www.testshib.org/register.html) of testshib configuration (make sure the filename is unique)
2.3. When editing SAML provider config (step 6 in Shibboleth Config guide), set `urn:oid:1.3.6.1.4.1.5923.1.1.1.6` as email hint (testshib does not send email, but sends [eduPersonPrincipalName](http://www.internet2.edu/products-services/trust-identity-middleware/mace-registries/internet2-object-identifier-oid-registrations/) which looks like an email and is just enough for testing purposes). Also, check "Autoprovision account" and "Skip email verification" checkboxes.
3. If you have aliases for lms in your environemnt (i.e. localhost:8000 and lms.local both point to LMS), make sure you're accessing signin view from the same hostname your exported metadata lists as `AssertionConsumerService[@Location]` attribute.
4. Click on the testshib SSO provider
![testshib](https://cloud.githubusercontent.com/assets/3330048/9581432/1dc3233a-5008-11e5-8047-14e247f22db3.png)
5. Use any username and password testhib provides (myself, alterego or superego)
6. Clicking "login" in Testhib should redirect to LMS and automatically provision an account for new user. Subsequent logins with same user should log him into an existing account
